torch/_subclasses/fake_tensor.py

```diff
@@ -803,6 +808,26 @@ class FakeTensor(torch.Tensor):
     fake_mode: "FakeTensorMode"
     constant: Optional[torch.Tensor]
 
+    # This memorizes the unbacked SymInt representing the number of nonzero
+    # elements in this tensor.  This is helpful if you do something like
+    # x[mask] and y[mask]; mask.nonzero() gets repeatedly called and should
+    # give a consistent unbacked SymInt.  It needs to be invalidated in the
+    # same way constant is.
+    # TODO: Generalize this as needed, e.g., into a trie of memos
+    _nonzero_memo: Optional[torch.SymInt]
+    _nonzero_memo_vc: Optional[int]
+
+    @property
+    def nonzero_memo(self):
+        if self._nonzero_memo is None:
+            return None
+        # Version counter based tracking isn't 100% sound but it's close
+        # enough
+        if self._nonzero_memo_vc != self._version:
+            self._nonzero_memo = None
+            return None
+        return self._nonzero_memo
```

Comment:
Would be nice to not add another invalidation mechanism. version_counter does not work in inference_mode, or with `resize_`. 

Unrelated: would be great to make version_counter always reliable, since it's very useful (as here), even if that meant  inference mode + pt2 incompatibility

Reply:
I agree with you in principle, but I can't really bring myself to care right now. I can file an issue for the backlog for if this actually becomes a problem.